lib/api/rule.go

```diff
@@ -128,3 +139,28 @@ type EntityRule struct {
 	// Protocol match in the Rule to be set to "tcp" or "udp".
 	NotPorts []numorstring.Port `json:"notPorts,omitempty" validate:"omitempty,dive"`
 }
+
+func combineNets(n *net.IPNet, nets []*net.IPNet) []*net.IPNet {
```

Comment:
Is there a requirement to support both `nets` and `net` on the same object?

Maybe we could simplify this by just validating that only one of `net` and `nets` is provided - might also be useful as a way to get users to drop using the deprecated field (they _need_ to stop using it in order to use the new feature) so that there's less pain when that field is dropped some day in the future.